langwatch/src/optimization_studio/components/properties/EndPropertiesPanel.tsx

```diff
@@ -1,14 +1,97 @@
-import type { Node } from "@xyflow/react";
+import { useUpdateNodeInternals, type Node } from "@xyflow/react";
 import type { End } from "../../types/dsl";
 import { BasePropertiesPanel } from "./BasePropertiesPanel";
+import { Text, Switch, HStack, Alert, AlertIcon } from "@chakra-ui/react";
+import { useEffect, useState } from "react";
+import { useWorkflowStore } from "~/optimization_studio/hooks/useWorkflowStore";
+
+export function EndPropertiesPanel({ node: initialNode }: { node: Node<End> }) {
+  const { node, setNode } = useWorkflowStore(
+    (state) => ({
+      node: state.nodes.find((n) => n.id === initialNode.id) as Node<End>,
+      setNode: state.setNode,
+    }),
+
+    // Add equality function to ensure proper updates
+    (prev, next) => prev.node === next.node
+  );
+
+  const updateNodeInternals = useUpdateNodeInternals();
+
+  const [isEvaluator, setIsEvaluator] = useState(
+    () =>
+      node.data.inputs?.length === 3 &&
+      node.data.inputs[0]?.identifier === "score"
+  );
```

Comment:
that is weird, this logic is super hardcoded! Shouldn't we use the `behave_as` instead? Otherwise if by accident I name my first output field "score" then all of a sudden its an evaluator